### ISSUE TITLE
Components accidentally reverted in Dec

### ISSUE THREAD

**Issue by SiccarPoint** [landlab/landlab#896]
My PR #770 appears to have been accidentally reverted away by @kbarnhart during another, unrelated reversion. This is a BIG PR, with two new components, tests, docs, etc. This seems like it happened the same day as my merge, with diff https://github.com/landlab/landlab/commit/adf20d52c44c8e006f2b4ac239a7af3e4b39fdaf#diff-3154fb794dbf9c168751fbf0ce117712.

Please put it back wholesale! My git skills are not delicate enough to do this myself. (@mcflugen can probably also solve this with minimal stress, I would guess?)

**Comment by kbarnhart** [comment]
@SiccarPoint I'm super sorry I accidentally did this. Will definitively fix. I'll also dig into why this happened to try and minimize the possibility of my doing it again. 

Once fixed, @mcflugen is presently the only person who can make a release. @mcflugen when you make the next release, will you add me as a conda-forge maintainer?

@mcflugen my approach to doing this (in a way that I know will work) is to cherrypick each commit individually. 

I've found [this StackOverflow](https://stackoverflow.com/questions/9853681/git-create-branch-from-range-of-previous-commits/9853814#9853814) very useful.

I've done the cherry-pick range approach before, but only on a smaller scale. I just tried it and it didn't work as I expected...

```
git cherry-pick 94044d8d43dc00bee3ae7b25f46e1e8d391aed5e^..015a982fbad114cb590110976c9c74395e59b049
```
Ended with 

```
error: could not apply 18d0185d... Rename the component file
hint: after resolving the conflicts, mark the corrected paths
hint: with 'git add <paths>' or 'git rm <paths>'
```

And almost all the files changed had to do with spatially variable precise.

**Comment by kbarnhart** [comment]
Yup. Not sure at all why I reverted that commit while working on PR #847. I think it was 100% accident. 🤦‍♀️ 🤦‍♀️ 🤦‍♀️ 🤦‍♀️ 🤦‍♀️ 🤦‍♀️ 🤦‍♀️ 🤦‍♀️ 🤦‍♀️ 🤦‍♀️ 🤦‍♀️ 🤦‍♀️ 🤦‍♀️ 🤦‍♀️ 🤦‍♀️ 🤦‍♀️ 🤦‍♀️ 🤦‍♀️ 🤦‍♀️ 🤦‍♀️ 🤦‍♀️ 🤦‍♀️ 🤦‍♀️ 🤦‍♀️ 🤦‍♀️ 🤦‍♀️

**Comment by kbarnhart** [comment]
Sorry for the open-close @SiccarPoint my mouse got over-active.

**Comment by kbarnhart** [comment]
Also I'm assigning myself to carefully look at #847 to see if I accidentally deleted anything else...

**Comment by kbarnhart** [comment]
This was addressed by #897. Closing.
